### PR TITLE
Upgrade package-lock.json to v3

### PR DESCRIPTION
**Pull request by mikeharder**
- Generated lockfile using Node20, which upgrades it to v3
- Generated using `npm install` rather than `npm update`, so there should be minimal changes to dependency versions
- Use `npm ci` instead of `npm install` to verify lockfile is updated
- List all packages after install

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -18,6 +18,19 @@ extends:
           name: $(LINUXPOOL)
           os: linux
         steps:
+          - task: NodeTool@0
+            displayName: 'Install Node'
+            inputs:
+              versionSpec: 20.x
+
+          - script: npm ci
```
Comment (mikeharder) [inline azure-pipelines.yml:26]: Ensures package-lock.json is updated

File: eng/test-steps.yml [hunk 2]
```diff
@@ -7,7 +7,12 @@ steps:
     displayName: 'Install Node'
     inputs:
       versionSpec: ${{ parameters.version }}
-  - script: npm install
+  - script: npm ci
+    displayName: 'npm ci'
+  - script: npm ls -a
+    displayName: 'npm ls -a'
+    condition: succeededOrFailed()
+    continueOnError: true
```
Comment (scbedd) [inline eng/test-steps.yml:15]: nit: intentional to kep a continueonError here?

**Review by scbedd — APPROVED** [review]
CI is green. Left minor nitpick feedback.

File: eng/test-steps.yml (hunk 2 above)
Comment (mikeharder) [reply eng/test-steps.yml:15]: Yes, we want to list the packages installed, even if `npm ci` fails.

**Pull request merged**